File: Tree/LearnYard/Binary_Tree_Preoder_traversal/Preorder.java
package Tree.LearnYard.Binary_Tree_Preoder_traversal;
import java.util.ArrayList;

class Node {
    int val;
    Node left, right;

    public Node(int val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

public class Preorder {
    public static void preorder_traversal(Node root, ArrayList<Integer> result) {
        if (root == null) {
            return;
        }
        result.add(root.val);
        preorder_traversal(root.left, result);
        preorder_traversal(root.right, result);
    }

    public static void main(String[] args) {
        Node root = new Node(1);
        root.left = new Node(2);
        root.right = new Node(3);
        root.left.left = new Node(4);
        root.left.right = new Node(5);

        ArrayList<Integer> result = new ArrayList<Integer>();
        preorder_traversal(root, result);

        for (int num : result) {
            System.out.print(num + " ");
        }
    }
}

File: Tree/LearnYard/Binary_Tree_Preoder_traversal/preorder.py
class Node:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def preorder(root, result):
    if root == None:
        return
    result.append(root.val)
    preorder(root.left, result)
    preorder(root.right, result)

# Creating a sample binary tree
#       1
#      / \
#     2   3
#    / \
#   4   5

root = Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.left.right = Node(5)

result = []
preorder(root, result)
print(result)